reference-implementation/lib/writable-stream.js

```diff
@@ -163,13 +163,37 @@ function WritableStreamAbort(stream, reason) {
   return promise;
 }
 
-function WritableStreamFinishAbort(stream) {
-  const error = new TypeError('Aborted');
-
+function WritableStreamError(stream, error) {
   stream._state = 'errored';
   stream._storedError = error;
 
-  WritableStreamRejectPromisesInReactionToError(stream);
+  // TODO: layering violation?
+  ResetQueue(stream._writableStreamController);
```

Comment:
We can introduce an interface and have different implementations for that. Yeah, virtual method as ricea said.

WritableStreamAbort() is touching the controller but through WritableStreamDefaultControllerAbort(). It's great if you abstract it into WritableStreamDefaultControllerError() e.g.